app/models/auth_sources/auth_source_ldap.rb

```diff
@@ -130,6 +131,23 @@ def users_in_group(name)
     ldap_con.user_list(name)
   end
 
+  def test_connection
+    result = {}
+    begin
+      Timeout::timeout(60) do
+        self.ldap_con.test
+      end
+      result[:success] = true
+      result[:message] = _("Test connection to LDAP server was successful.")
+    rescue Timeout::Error, Net::LDAP::Error, StandardError   => exception
+      foreman_exception = Foreman::Exception.new exception.message
+      result[:success] = false
+      result[:error_class] = exception.class.name
+      result[:message] = "#{foreman_exception.code}: #{_(exception.message)}"
```

Comment:
Not sure that `_()` is any use here, unless you're also able to extract the original messages, leave it out.

Rather than constructing a Foreman::Exception and using its code implementation locally only, should this just raise a Foreman::Exception, or rather, a wrapped exception?